src/core/runtime.cpp

```diff
@@ -978,12 +978,10 @@ void validateConnection(ValidationContext &ctx) {
         shards::arrayPush(data.shared, type);
       }
     }
-    if (!ctx.wire || !ctx.wire->pure) {
-      // and inherited
-      for (auto &info : ctx.inherited) {
-        for (auto &type : info.second) {
-          shards::arrayPush(data.shared, type);
-        }
+    // and inherited
```

Comment:
this likely is no good? are we sure @guusw ?

Reply:
Will add comment to clarify